proposals/secret-management-for-discovery-handler.md

```diff
@@ -0,0 +1,115 @@
+```DiscoveryHandler``` service (the secret/data will be part of the ```DiscoverRequest```).  This additional data, named 
+‘discover property’ will in the format of string key-value pair list.  Once DH received the discovery properties and 
+discovery details, it should be able to perform discovery tasks.  The Agent will watch any discovery property changes 
+from the discovery property data source and re-issue a ```DiscoverRequest``` if any of the data changed.
+
+The Discovery properties will be specified as part of the Akri configuration, here is an example.  The 
+```discoveryProperties``` section is newly added for specifying the secret information. Current plan is to support 
+both k8s ```secret``` and k8s ```configMap```. The ```discoveryProperties``` is optional.  If 
+```discoveryProperties``` is not specified, it falls back to what we have currently, just ```discoveryDetails``` 
+passed to DH.
+
+The key-value pair in ```discoveryProperties``` is a list and can be as many entries as needed.
+
+The key-value pair can be optional. The ```optional``` attribute in the ```discoveryProperties``` entry is default to 
+```false```, it means if the key doesn't exist in the ```secret``` (or ```configMap```), the Configuration deployment 
+will fail.  If ```optional``` is ```true``` and the secret key doesn't exist, Agent will not add the entry to the list 
+passed to DH, the Configuration deployment will success.
+
+## Secret data changes after deploying the Akri Configuration
+Current plan Akri Agent will not monitor the secret data changes after the Akri Configuration is deployed.  If the 
+content of ```discoveryProperties``` changed from the data sources after Configuration deployed, users need to 
+manually update the Akri Configuration.  If necessary, we can make improvement to have Akri Agent monitors the secret 
+data source and re-issue discoverRequest if the data source changed.  For example, username and password are stored in 
+a k8s secret and are specified in the ```discoverProperties``` as mandatory in an Akri Configuration.  After the 
+configuration deployed, Agent can watch any change to these two properties in k8s secret and re-issue discoverRequest 
+if they changed.  If the mandatory property is deleted from the data store, the configuration is considered invalid, 
+and Akri Agent will revoke the discoveryRequest from DH.
+
+```yaml
+apiVersion: akri.sh/v0
+kind: Configuration
+metadata:
+  name: akri-onvif-video
+spec:
+  discoveryHandler:
+    name: onvif
+    discoveryDetails: |+
+        ipAddresses:
+          action: Exclude
+          items:
+          - 10.0.0.1
+          - 10.0.0.2
+        macAddresses:
+          action: Exclude
+          items: []
+        scopes:
+          action: Include
+          items:
+          - onvif://www.onvif.org/name/GreatONVIFCamera
+          - onvif://www.onvif.org/name/AwesomeONVIFCamera
+        discoveryTimeoutSeconds: 2
+    discoveryProperties:
```

Comment:
Why a generalized name of `discoveryProperties` instead of something more specific and narrowly scoped like `deviceCredentials`?

Reply:
on purpose to use a generic name, users can use this field to pass information required for DH, not just credentials.